### PR TITLE
add FranklinTempleton RWA adapter

### PR DESCRIPTION
**Pull request by 0xpeluche**
---
##### Name (to be shown on DefiLlama):
Franklin Templeton Benji Investments

##### Website Link:
https://www.franklintempleton.com/investments/options/money-market-funds/products/29386/SINGLCLASS/franklin-on-chain-u-s-government-money-fund/FOBXX

##### Current TVL:
400M

##### Chain:
Stellar, Polygon

##### Short Description (to be shown on DefiLlama):
Franklin OnChain U.S. Government Money Fund

##### Token address and ticker if any:
Stellar: BENJI, GBHNGLLIE3KWGKCHIKMHJ5HVZHYIK7WTBE4QF5PLAKL4CJGSEU7HZIW5
Polygon: BENJI, 0x408a634b8a8f0de729b48574a3a7ec3fe820b00a

##### Category (full list at https://defillama.com/categories) *Please choose only one:
RWA

##### methodology (what is being counted as tvl, how is tvl being calculated):
TVL counts total Assets mint

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/franklinTempleton-finance exports TVL: 
        
 
 
    stellar                   399.61 M
    polygon                   2.13 M
    
    total                    401.74 M

**Comment by g1nt0ki** [comment]
hmm, net assets seems to be 348M so maybe part of the of the supply is not yet issued?

File: projects/franklinTempleton-finance/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    const adjustedSupply = supply * Math.pow(10, USDC_DECIMALS) / Math.pow(10, toml_info.decimals)
+    return api.add(ADDRESSES.ethereum.USDC, adjustedSupply, { skipChain: true })
+}
+
+const polygonTvl = async (api) => {
+  const [decimals, totalSupply] = await Promise.all([
+        api.call({target: BENJI_POLYGON, abi:'erc20:decimals'}),
+        api.call({target: BENJI_POLYGON, abi:'erc20:totalSupply'})
+  ])
+  const adjustedSupply = totalSupply * Math.pow(10, USDC_DECIMALS) / Math.pow(10, decimals)
```
Comment (0xngmi) [inline projects/franklinTempleton-finance/index.js:20]: there's a toUSDTBalances() function you can use for these cases, much simpler than this

**Comment by 0xpeluche** [comment]
yea on the website there is 
image

but last update was at the end of may and then : 

image

File: projects/franklinTempleton-finance/index.js (hunk 1 above)
Comment (0xpeluche) [reply projects/franklinTempleton-finance/index.js:20]: Oh, thank you, that's good to know.

**Comment by llamatester** [comment]
The adapter at projects/franklinTempleton-finance exports TVL: 
        
 
 
    stellar                   398.76 M
    polygon                   2.12 M
    
    total                    400.88 M

**Pull request merged**